### PR TITLE
Add PPOM testing section

### PR DESCRIPTION
**Pull request by seaona**
## Description
This PR tackles these 2 needs:

- We need an easy and fast way to manually test the ppom module.
- We need to automate the testcases related to the ppom flows.

For that, a new section on the test dapp has been created, where we can trigger all the possible warnings fired by the ppom module.

### Notes for Testing
Bear in mind the following points, in case you want to try the new dapp functionalities:
- at the moment ppom module is only supported on Mainnet, with no other RPCs than Infura, with special Infura Id (ask me if you need it). In the future, support is going to be extended to more RPCs (in the next ppom version) and more networks/chainIds. 
- at the moment, ppom module is only available in flask builds (with proper envars setup `BLOCKAID_FILE_CDN` and `BLOCKAID_PUBLIC_KEY`)
- you need to activate the functionality under Experimental Settings / Blockaid
- you need to have a small amount of ETH balance for malicious transactions and some amount of BUSD for the ERC20 related ones (once different RPCs are supported, you can easily bypass this by using a Tenderly devnet and seeding your account (see more in the Workshop recording about this topic -[ Configurable Forks & Contract Storage Manipulation](https://www.notion.so/f649ecf027cc41db858d3a3574fe3a99?v=da4af5e48f0e44d7b4536c4e2e911f36))
- you don't need any balance for Signatures (Malicious Permit/Trade Order/ Seaport)

https://github.com/MetaMask/test-dapp/assets/54408225/0704c09a-a2bc-4642-b783-24a6c5276cc4

### Extra Info
- Fixes https://github.com/MetaMask/MetaMask-planning/issues/1189
- Missing Send ERC721 and SetApprovalForAll cases (will be done in a subsequent PR once current blockers are sorted out)

## Screenshots
Screenshot from 2023-08-24 17-58-47

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1339,8 +1452,8 @@ const initialize = async () => {
   siweBadAccount.onclick = async () => {
     const domain = window.location.host;
     const from = '0x0000000000000000000000000000000000000000';
-    const siweMessageBadDomain = `${domain} wants you to sign in with your Ethereum account:\n${from}\n\nI accept the MetaMask Terms of Service: https://community.metamask.io/tos\n\nURI: https://${domain}\nVersion: 1\nChain ID: 1\nNonce: 32891757\nIssued At: 2021-09-30T16:25:24.000Z\nResources:\n- ipfs://Qme7ss3ARVgxv6rXqVPiikMJ8u2NLgmgszg13pYrDKEoiu\n- https://example.com/my-web2-claim.json`;
-    siweSign(siweMessageBadDomain);
+    const siweMessageBadAccount = `${domain} wants you to sign in with your Ethereum account:\n${from}\n\nI accept the MetaMask Terms of Service: https://community.metamask.io/tos\n\nURI: https://${domain}\nVersion: 1\nChain ID: 1\nNonce: 32891757\nIssued At: 2021-09-30T16:25:24.000Z\nResources:\n- ipfs://Qme7ss3ARVgxv6rXqVPiikMJ8u2NLgmgszg13pYrDKEoiu\n- https://example.com/my-web2-claim.json`;
```
Comment (seaona) [inline src/index.js:1455]: Small unrelated fix: make the const name match the content, so it's not miss-leading

**Review by jpuri — APPROVED** [review]
